### PR TITLE
Catch auxdata decode error when checking for perfect match

### PR DESCRIPTION
**Pull request by sealer3**
This helps when verifying contracts that don't have the CBOR metadata block at the end; currently, an error is thrown rather than making it a partial match.

### CODE REVIEW

**Review by marcocastignoli — APPROVED** [review]
You are right, lets just wait for @kuzdogan's approval, but for me we can merge this

**Comment by kuzdogan** [comment]
Thank you!

**Pull request merged**
